### ISSUE TITLE
search box missing in newest version

### ISSUE THREAD

**Issue by j0hannes** [JabRef/jabref#3505]
- [x] I have tested the latest master version from http://builds.jabref.org/master/ and the problem persists

JabRef version 4.1-dev

Steps to reproduce:

1. look for search box: it isn't there (see screenshots)

old version (2017-11-10):
![screenshot_20171209_121417](https://user-images.githubusercontent.com/3944397/33795143-a0c89ad2-dcdb-11e7-9942-e378ac03113d.png)

newest version:
![screenshot_20171209_121440](https://user-images.githubusercontent.com/3944397/33795144-a0efefc4-dcdb-11e7-817c-6d2daedd0aa1.png)

**Comment by Siedlerchr** [comment]
Xubuntu 16.04,
javax.swing.plaf.metal.MetalLookAndFeel
![image](https://user-images.githubusercontent.com/320228/33796670-8a29788c-dcf9-11e7-8376-4cdd3debf925.png)

**Comment by j0hannes** [comment]
I switched to javax.swing.plaf.metal.MetalLookAndFeel now (under Arch now), but it looks different to your style:

![screenshot_20171209_165315](https://user-images.githubusercontent.com/3944397/33797068-86dde674-dd01-11e7-8ffa-405851de1bfd.png)

**Comment by lenhard** [comment]
@j0hannes You're not listing anything about your Java runtime. Might as well be an outdated Java runtime or outdated JavaFX.

**Comment by j0hannes** [comment]
```
JabRef 4.1-dev
Linux 4.9.66-1-lts amd64 
Java 1.8.0_144
```

**Comment by lenhard** [comment]
> or outdated JavaFX.

**Comment by j0hannes** [comment]
where do I see that?

**Comment by tobiasdiez** [comment]
You use ArchLinux, right? Then `pacman -Syu` should upgrade all packages (and hopefully also JavaFX). Do you use OpenJDK/OpenJFX or Oracle (`java -version`)?

**Comment by j0hannes** [comment]
I think this is the package: `extra/java-openjfx 8.u172-1 [installed]`

`java -version` prints
```
openjdk version "9"
OpenJDK Runtime Environment (build 9+181)
OpenJDK 64-Bit Server VM (build 9+181, mixed mode)
```
but for JabRef I commented in the line in the `jabref` shell script that explictely makes it use version 8:
`export PATH=/usr/lib/jvm/java-8-openjdk/jre/bin/:$PATH`

`/usr/lib/jvm/java-8-openjdk/jre/bin/java -version` gives
```
openjdk version "1.8.0_144"
OpenJDK Runtime Environment (build 1.8.0_144-b01)
OpenJDK 64-Bit Server VM (build 25.144-b01, mixed mode)
```

**Comment by tobiasdiez** [comment]
Thanks! That actually looks good and up-to-date. Is the SearchBar the only problem? Do the other JavaFX windows like groups, entry editor and about dialog work? Do you get an exception or some unusual print-out in the about > error console?

**Comment by j0hannes** [comment]
There it is!
```
Uncaught exception occurred in Thread[JavaFX Application Thread,5,main]
java.lang.NullPointerException
	at org.jabref.gui.search.GlobalSearchBar.lambda$new$5(GlobalSearchBar.java:190)
	at com.sun.javafx.application.PlatformImpl.lambda$null$5(PlatformImpl.java:295)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$6(PlatformImpl.java:294)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at com.sun.glass.ui.gtk.GtkApplication.lambda$null$5(GtkApplication.java:139)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by tobiasdiez** [comment]
Ok this is helpful! Which snapshot are you using (under Help > About, should be of the form JabRef 4.1-dev--snapshot--2017-11-24--master--8025c698a). The most [recent](https://builds.jabref.org/master/) is 2017-12-08 and I exactly changed something in this line. So it would be helpful to know if you run the most recent or a previous one. Thanks!

**Comment by j0hannes** [comment]
According to that dialog it's `JabRef 4.1-dev`. I just updated to the latest git version and the NullPointerException is still there.

**Comment by tobiasdiez** [comment]
At which line? 190 or 191?

**Comment by j0hannes** [comment]
190

**Comment by tobiasdiez** [comment]
Thanks! I think I found the problem and a fix is in the pipeline #3512.
